### DISCUSSION TITLE
UCB CS169: software engineering - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#107] [Announcements]
# UCB CS169: software engineering - CS自学指南

CS自学指南

https://csdiy.wiki/%E8%BD%AF%E4%BB%B6%E5%B7%A5%E7%A8%8B/CS169/

**Comment by 2018hahazhufeng** [comment]
课程好像关闭enroll了

**Reply by dengyunsheng250** [reply]
那这个课程还能做不？

**Reply by D-shuoshuo** [reply]
好像不行

**Reply by hychen11** [reply]
那该怎么学习呢？

**Reply by sbwcwso** [reply]
是不是作业的评分之类的做不了了？

**Reply by MTDickens** [reply]
有一个 2018 年版本的，里面有作业、习题之类的，而且免费。

网址：https://learning.edx.org/course/course-v1:BerkeleyX+CS169.1X+2T2018/home

**Reply by morosm** [reply]
课程的视频都消失了- -,

**Reply by zxhd863943427** [reply]
补一个第二部分的课程：https://learning.edx.org/course/course-v1:BerkeleyX+CS169.2x+1T2018/home

**Comment by houdiniping** [comment]
> 课程的视频都消失了- -,

官网往下翻，在最后面

**Comment by slender-nonbuckler** [comment]
还是可以学的 视频和课本看 https://saasbook.info/
作业都存储在https://github.com/saasbook/    每节课的作业查书中的CHIPS（cs169作业的别称）。只有一个问题是作业源代码是用ruby 2.6写的，目前在维护的ruby 是3.0以上的，需要自己调整一下代码相关内容或者下载老版本。

**Comment by f1a3h** [comment]
ESaaS Instructors' Guide: https://teach.saasbook.info

**Reply by morosm** [reply]
非常感谢!

**Reply by ShaneHeX** [reply]
这门课不适合初学者，还是 MIT 6.031 这门课会更适合入门一些

**Comment by Anxi7** [comment]
在 YouTube 上有最新的课程视频 https://www.youtube.com/@saasbook

**Reply by sbwcwso** [reply]
请问有最新版的作业不？

**Comment by bitterlucky** [comment]
是不是现在做不了了
